config/v1/0000_10_config-operator_01_ingress.crd.yaml

```diff
@@ -53,6 +53,54 @@ spec:
                 appsDomain may be modified after installation. This assumes a new
                 ingresscontroller has been setup with a wildcard certificate.
               type: string
+            componentRoutes:
+              description: "componentRoutes is an optional list of routes configured
+                by the cluster-admin to override the default hostname and serving
+                certificates. The namespace and name of each componentRoute should
+                match an existing componentRoute defined in the status. \n To determine
+                the set of configurable Routes, look at namespace and name of entries
+                in the .status.componentRoutes, where participating operators place
+                current route status."
+              type: array
+              items:
+                description: ComponentRouteSpec allows for configuration of a route's
+                  hostname and serving certificate.
+                type: object
+                required:
+                - hostname
+                - name
+                - namespace
+                - servingCertKeyPairSecret
+                properties:
+                  hostname:
+                    description: hostname is the hostname that should be supported
+                      by the route.
+                    type: string
+                    format: hostname
+                    minLength: 1
+                  name:
+                    description: name is the logical name of the route to customize.
+                    type: string
+                    minLength: 1
+                  namespace:
+                    description: namespace is the namespace of the route to customize.
+                      It must be a real namespace. Using an actual namespace ensures
+                      that no two components will conflict and the same component
+                      can be installed multiple times.
+                    type: string
+                    pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
```

Comment:
This pattern wont match namespace names that contain `-`, eg. `console-operator`.  Also wondering if we should not be more restrictive and have a strict check for `openshift-` prefix here since all the [mentioned](https://github.com/openshift/enhancements/pull/577#issue-551856503) namespaces are starting with the prefix.